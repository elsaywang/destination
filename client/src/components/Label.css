:local(.inline-block) {
    display: inline-block;
    margin: 0 10px;
    height: 72px;
    vertical-align: bottom;
}

:local(.inline-block + .inline-block) {
    margin-left: 0;
}

:local(.label) {
    display: block;
    margin: 10px 0;
    font-size: 0.875em;
    vertical-align: middle;
    transition: color 150ms ease-in-out;
}
