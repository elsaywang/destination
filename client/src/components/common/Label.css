:local(.inline-block) {
    display: inline-block;
    margin: 0 10px;
    vertical-align: bottom;
}

:local(.inline-block + .inline-block) {
    margin-left: 0;
}

:local(.label) {
    display: block;
    margin: 10px 0;
    font-size: 0.875em;
    transition: color 150ms ease-in-out;
}
