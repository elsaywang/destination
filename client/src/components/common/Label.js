import React from 'react';
import classNames from 'classnames';
import styles from './Label.css';

function Label(props) {
    return (
        <span className={classNames(styles['inline-block'], props.className)}>
            <label htmlFor={props.labelFor} className={styles.label} style={props.style}>
                {props.value}
            </label>
            {props.children}
        </span>
    );
}

export default Label;
