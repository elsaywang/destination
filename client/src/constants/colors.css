/* TODO: utilize https://git.corp.adobe.com/Spectrum/spectrum-dna for css vars */

@value borderGrey: #CCC;
