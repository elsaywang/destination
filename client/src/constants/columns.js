/**
 * All column keys
 */
export const columnKeys = {
    keyValuePairs: 'keyValuePairs',
    keyName: 'keyName',
    signalType: 'signalType',
    signalSource: 'signalSource',
    totalCount: 'totalCount',
    percentageChange: 'percentageChange',
    includedInTraits: 'includedInTraits',
};

/**
 * Individual columns
 */
const keyValuePairs = {
    title: 'Key-Value Pair',
    key: columnKeys.keyValuePairs,
    minWidth: 150,
    sortable: true,
};
const keyName = {
    title: 'Key Name',
    key: columnKeys.keyName,
    width: 200,
};
const signalType = {
    title: 'Signal Type',
    key: columnKeys.signalType,
    width: 140,
};
const signalSource = {
    title: 'Signal Source',
    key: columnKeys.signalSource,
    width: 180,
};

const reportSuite = {
    title: 'Report Suite',
    key: columnKeys.signalSource,
    width: 180,
};

const onboardedRecord = {
    title: 'Onboarded Record',
    key: columnKeys.signalSource,
    width: 180,
};
/**
 * The "Total Counts", "Event Fires", and "Total Records" column will
 * all use the `totalCount` (no 's') column key. They're all used to display the
 * `totalCount` property of a signal, but we use different labels depending on
 * the selected signal type filter.
 *
 * All - "Total Counts"
 * Adobe Analytics - "Event Fires"
 * Actionable Log Files - "Event Fires"
 * General Online Data - "Event Fires"
 * Onboarded Records - "Total Records"
 */
const totalCounts = {
    title: 'Total Counts',
    key: columnKeys.totalCount,
    width: 88,
    sortable: true,
};
const eventFires = {
    title: 'Event Fires',
    key: columnKeys.totalCount,
    width: 88,
    sortable: true,
};
const totalRecords = {
    title: 'Total Records',
    key: columnKeys.totalCount,
    width: 88,
    sortable: true,
};
const percentageChange = {
    title: 'Percentage Change',
    key: columnKeys.percentageChange,
    width: 165,
    sortable: true,
};
const includedInTraits = {
    title: 'Included In Traits',
    key: columnKeys.includedInTraits,
    width: 155,
};

/**
 * Column groups by signal type
 */
export const allSignalsColumns = [
    keyValuePairs,
    signalType,
    signalSource,
    totalCounts,
    percentageChange,
    includedInTraits,
];

export const analyticsColumns = [
    keyValuePairs,
    signalType,
    reportSuite,
    eventFires,
    percentageChange,
    includedInTraits,
];

export const advancedAnalyticsColumns = [
    keyValuePairs,
    keyName,
    eventFires,
    percentageChange,
    includedInTraits,
];

export const actionableLogFilesColumns = [
    keyValuePairs,
    signalType,
    eventFires,
    percentageChange,
    includedInTraits,
];

export const generalOnlineDataColumns = [...actionableLogFilesColumns];

export const onboardedRecordsColumns = [
    keyValuePairs,
    signalType,
    onboardedRecord,
    totalRecords,
    percentageChange,
    includedInTraits,
];

export const sortColumns = [
    totalCounts,
    percentageChange,
    {
        ...keyValuePairs,
        title: 'Key Name',
    },
];

/**
 * All columns
 */
export const columns = [
    keyValuePairs,
    keyName,
    signalType,
    signalSource,
    reportSuite,
    onboardedRecord,
    totalCounts,
    eventFires,
    totalRecords,
    percentageChange,
    includedInTraits,
];
