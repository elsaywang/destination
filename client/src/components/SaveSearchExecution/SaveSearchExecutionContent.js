import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Label from '../../components/common/Label';
import Textfield from '@react/react-spectrum/Textfield';
import Checkbox from '@react/react-spectrum/Checkbox';
import RadioGroup from '@react/react-spectrum/RadioGroup';
import Radio from '@react/react-spectrum/Radio';
import Select from '@react/react-spectrum/Select';
import styles from './SaveSearchExecution.css';
import {
    textFieldLabelName,
    textFieldPlaceHolder,
    saveSearch,
    checkBoxLabel,
    defaultSorting,
} from './saveSearchExecutionMessages.js';
import {
    descending,
    sortingOptions,
    radioGroupOptions,
    defaultSortingKey,
} from '../../constants/sortOptions';

class SaveSearchExecutionContent extends Component {
    state = {
        isTrackInDashboardChecked: false,
    };

    toggleCheckbox = isTrackInDashboardChecked => {
        this.setState({ isTrackInDashboardChecked });
        this.props.onTrackResultInDashboardChange(isTrackInDashboardChecked);
    };

    handleRadioChange = value => {
        this.props.onSortingOrderChange(value === descending);
    };

    render() {
        return (
            <div data-test="save-this-search-dialog-content">
                <Label value={textFieldLabelName} labelFor={saveSearch}>
                    <Textfield
                        data-test="save-this-search-name-field"
                        className={styles.contentTextField}
                        placeholder={textFieldPlaceHolder}
                        onChange={this.props.onSaveSearchNameChange}
                        quiet
                    />
                </Label>
                <div className={styles.contentCheckBox}>
                    <Checkbox
                        data-test="save-this-search-checkbox"
                        label={checkBoxLabel}
                        onChange={this.toggleCheckbox}
                        checked={this.state.isTrackInDashboardChecked}
                    />
                </div>
                <Label value={defaultSorting} labelFor={defaultSorting}>
                    <Select
                        options={sortingOptions}
                        onChange={this.props.onDefaultSortingChange}
                        defaultValue={defaultSortingKey}
                    />
                    <div className={styles.contentRadioGroup}>
                        <RadioGroup>
                            {radioGroupOptions.map(({ label, value }) => (
                                <Radio
                                    key={value}
                                    {...{ label, value }}
                                    checked={value === descending}
                                    onChange={this.handleRadioChange}
                                />
                            ))}
                        </RadioGroup>
                    </div>
                </Label>
            </div>
        );
    }
}

SaveSearchExecutionContent.propTypes = {
    onSaveSearchNameChange: PropTypes.func,
    onTrackResultInDashboardChange: PropTypes.func,
    onDefaultSortingChange: PropTypes.func,
    onSortingOrderChange: PropTypes.func,
};
export default SaveSearchExecutionContent;
