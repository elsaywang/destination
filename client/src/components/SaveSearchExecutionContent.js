import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import Label from '../components/common/Label';
import Textfield from '@react/react-spectrum/Textfield';
import Checkbox from '@react/react-spectrum/Checkbox';
import SaveSearchSortingCriteria from './SaveSearchSortingCriteria';
import styles from './SaveSearchExecution.css';
import {
    textFieldLabelName,
    textFieldPlaceHolder,
    saveThisSearch,
    checkBoxLabel,
    defaultSorting,
    radioGroupOptions,
    sortingOptions,
} from '../utils/saveSearch';

class SaveSearchExecutionContent extends Component {
    constructor() {
        super();
        this.state = {
            isBoxChecked: false,
        };
    }

    toggleCheckbox = () => {
        this.setState({ isBoxChecked: !this.state.isBoxChecked });
        this.props.onTrackResultInDashBoardChange(!this.state.isBoxChecked);
    };

    handleRadioChange = value => {
        this.props.onSortingOrderChange(value === 'descending');
    };

    executionForm = () => {
        return (
            <Fragment>
                <Label value={textFieldLabelName} labelFor={saveThisSearch}>
                    <Textfield
                        style={{ minWidth: 230 }}
                        placeholder={textFieldPlaceHolder}
                        onChange={this.props.onSaveSearchNameChange}
                        quiet
                    />
                </Label>
                <div className={styles.contentCheckBox}>
                    <Checkbox
                        label={checkBoxLabel}
                        onChange={this.toggleCheckbox}
                        checked={this.state.isBoxChecked}
                    />
                </div>
                {this.state.isBoxChecked && (
                    <Label value={defaultSorting} labelFor={defaultSorting}>
                        <SaveSearchSortingCriteria
                            selectOptions={sortingOptions}
                            radioGroupOptions={radioGroupOptions}
                            onDefaultSortingChange={this.props.onDefaultSortingChange}
                            handleRadioChange={this.handleRadioChange}
                            radioCheckedValue="descending"
                            radioGroupStyle={styles.contentRadioGroup}
                        />
                    </Label>
                )}
            </Fragment>
        );
    };
    render() {
        return this.executionForm();
    }
}

SaveSearchExecutionContent.propTypes = {
    onSaveSearchNameChange: PropTypes.func,
    onTrackResultInDashBoardChange: PropTypes.func,
    onDefaultSortingChange: PropTypes.func,
    onSortingOrderChange: PropTypes.func,
};
export default SaveSearchExecutionContent;
