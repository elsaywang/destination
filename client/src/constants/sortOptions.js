import { sortColumns } from './columns';

export const ascending = 'ascending';
export const descending = 'descending';

export const sortingOptions = sortColumns.map(({ title, key }) => ({
    label: title,
    value: key,
}));

export const radioGroupOptions = [
    { label: 'Descending', value: descending },
    { label: 'Ascending', value: ascending },
];
