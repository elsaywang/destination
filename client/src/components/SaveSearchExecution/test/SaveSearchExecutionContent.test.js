import React from 'react';
import { shallow } from 'enzyme';
import SaveSearchExecutionContent from '../SaveSearchExecutionContent';
import Label from '../../../components/common/Label';
import Textfield from '@react/react-spectrum/Textfield';
import Checkbox from '@react/react-spectrum/Checkbox';
import Select from '@react/react-spectrum/Select';
import RadioGroup from '@react/react-spectrum/RadioGroup';
import Radio from '@react/react-spectrum/Radio';
import { descending, sortingOptions, radioGroupOptions } from '../../../constants/sortOptions';
import {
    textFieldLabelName,
    textFieldPlaceHolder,
    saveThisSearch,
    checkBoxLabel,
    defaultSorting,
} from '../saveSearchExecutionMessages';

describe('<SaveSearchExecutionContent /> component', () => {
    const mockFn = jest.fn();
    const wrapper = shallow(
        <SaveSearchExecutionContent
            onSaveSearchNameChange={mockFn}
            onTrackResultInDashboardChange={mockFn}
            onDefaultSortingChange={mockFn}
            onSortingOrderChange={mockFn}
        />,
    );
    describe('rendering', () => {
        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it(`renders <Label/> with correct label and value and it's child <Textfield> with correct placeholder name`, () => {
            expect(wrapper.find(Label).exists()).toBeTruthy();
            expect(wrapper.find(Label).props().value).toEqual(textFieldLabelName);
            expect(wrapper.find(Label).props().labelFor).toEqual(saveThisSearch);
            expect(
                wrapper
                    .find(Label)
                    .children()
                    .find(Textfield),
            ).toBeTruthy();
            expect(
                wrapper
                    .find(Label)
                    .children()
                    .find(Textfield)
                    .props().placeholder,
            ).toEqual(textFieldPlaceHolder);
        });
        it(`renders <Checkbox/> with correct label and default is unchecked`, () => {
            expect(wrapper.state('isTrackInDashboardChecked')).toBeFalsy();
            expect(wrapper.find(Checkbox).exists()).toBeTruthy();
            expect(wrapper.find(Checkbox).props().label).toEqual(checkBoxLabel);
            expect(wrapper.find(Checkbox).props().checked).toBeFalsy();
        });
    });

    describe('when Checkbox is checked', () => {
        beforeAll(() => {
            wrapper.instance().toggleCheckbox(true);
        });
        it('state.isTrackInDashboardChecked is updated to true ', () => {
            expect(wrapper.state('isTrackInDashboardChecked')).toBeTruthy();
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <Label/> with correct label/value and children of <Select/> and <RadioGroup/> including 2 <Radio/>', () => {
            wrapper.setState({ isTrackInDashboardChecked: true });
            expect(wrapper.find(Select)).toBeTruthy();
            expect(wrapper.find(RadioGroup).exists()).toBeTruthy();
            expect(wrapper.find(Radio).length).toBe(2);
        });

        it(`renders <RadioGroup/> including 2 <Radio/> with correct label and value`, () => {
            radioGroupOptions.map(({ label, value }, index) => {
                expect(
                    wrapper
                        .find(Radio)
                        .at(index)
                        .props().label,
                ).toEqual(label);
                expect(
                    wrapper
                        .find(Radio)
                        .at(index)
                        .props().value,
                ).toEqual(value);
                expect(
                    wrapper
                        .find(Radio)
                        .at(index)
                        .props().checked,
                ).toEqual(value === descending);
            });
        });
    });
});
