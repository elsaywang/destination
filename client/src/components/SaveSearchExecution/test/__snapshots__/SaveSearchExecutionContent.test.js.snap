// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SaveSearchExecutionContent /> component rendering matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SaveSearchExecutionContent
    onDefaultSortingChange={[MockFunction]}
    onSaveSearchNameChange={[MockFunction]}
    onSortingOrderChange={[MockFunction]}
    onTrackResultInDashboardChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Label
          labelFor="Save This Search"
          value="Name"
        >
          <Textfield
            className={undefined}
            data-test="save-this-search-name-field"
            disabled={false}
            invalid={false}
            onChange={[MockFunction]}
            placeholder="Enter a name for this saved search"
            quiet={true}
            readOnly={false}
            required={false}
          />
        </Label>,
        <div
          className={undefined}
        >
          <Checkbox
            checked={false}
            indeterminate={false}
            label="Track this search result in dashboard"
            onChange={[Function]}
          />
        </div>,
        false,
      ],
      "data-test": "save-this-search-dialog-content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Textfield
            className={undefined}
            data-test="save-this-search-name-field"
            disabled={false}
            invalid={false}
            onChange={[MockFunction]}
            placeholder="Enter a name for this saved search"
            quiet={true}
            readOnly={false}
            required={false}
          />,
          "labelFor": "Save This Search",
          "value": "Name",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": undefined,
            "data-test": "save-this-search-name-field",
            "disabled": false,
            "invalid": false,
            "onChange": [MockFunction],
            "placeholder": "Enter a name for this saved search",
            "quiet": true,
            "readOnly": false,
            "required": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Checkbox
            checked={false}
            indeterminate={false}
            label="Track this search result in dashboard"
            onChange={[Function]}
          />,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": false,
            "indeterminate": false,
            "label": "Track this search result in dashboard",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Label
            labelFor="Save This Search"
            value="Name"
          >
            <Textfield
              className={undefined}
              data-test="save-this-search-name-field"
              disabled={false}
              invalid={false}
              onChange={[MockFunction]}
              placeholder="Enter a name for this saved search"
              quiet={true}
              readOnly={false}
              required={false}
            />
          </Label>,
          <div
            className={undefined}
          >
            <Checkbox
              checked={false}
              indeterminate={false}
              label="Track this search result in dashboard"
              onChange={[Function]}
            />
          </div>,
          false,
        ],
        "data-test": "save-this-search-dialog-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Textfield
              className={undefined}
              data-test="save-this-search-name-field"
              disabled={false}
              invalid={false}
              onChange={[MockFunction]}
              placeholder="Enter a name for this saved search"
              quiet={true}
              readOnly={false}
              required={false}
            />,
            "labelFor": "Save This Search",
            "value": "Name",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": undefined,
              "data-test": "save-this-search-name-field",
              "disabled": false,
              "invalid": false,
              "onChange": [MockFunction],
              "placeholder": "Enter a name for this saved search",
              "quiet": true,
              "readOnly": false,
              "required": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Checkbox
              checked={false}
              indeterminate={false}
              label="Track this search result in dashboard"
              onChange={[Function]}
            />,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "indeterminate": false,
              "label": "Track this search result in dashboard",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<SaveSearchExecutionContent /> component when Checkbox is checked state.isTrackInDashboardChecked is updated to true  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SaveSearchExecutionContent
    onDefaultSortingChange={
      [MockFunction] {
        "calls": Array [
          Array [
            true,
          ],
        ],
      }
    }
    onSaveSearchNameChange={
      [MockFunction] {
        "calls": Array [
          Array [
            true,
          ],
        ],
      }
    }
    onSortingOrderChange={
      [MockFunction] {
        "calls": Array [
          Array [
            true,
          ],
        ],
      }
    }
    onTrackResultInDashboardChange={
      [MockFunction] {
        "calls": Array [
          Array [
            true,
          ],
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Label
          labelFor="Save This Search"
          value="Name"
        >
          <Textfield
            className={undefined}
            data-test="save-this-search-name-field"
            disabled={false}
            invalid={false}
            onChange={
              [MockFunction] {
                "calls": Array [
                  Array [
                    true,
                  ],
                ],
              }
            }
            placeholder="Enter a name for this saved search"
            quiet={true}
            readOnly={false}
            required={false}
          />
        </Label>,
        <div
          className={undefined}
        >
          <Checkbox
            checked={false}
            indeterminate={false}
            label="Track this search result in dashboard"
            onChange={[Function]}
          />
        </div>,
        false,
      ],
      "data-test": "save-this-search-dialog-content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Textfield
            className={undefined}
            data-test="save-this-search-name-field"
            disabled={false}
            invalid={false}
            onChange={
              [MockFunction] {
                "calls": Array [
                  Array [
                    true,
                  ],
                ],
              }
            }
            placeholder="Enter a name for this saved search"
            quiet={true}
            readOnly={false}
            required={false}
          />,
          "labelFor": "Save This Search",
          "value": "Name",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": undefined,
            "data-test": "save-this-search-name-field",
            "disabled": false,
            "invalid": false,
            "onChange": [MockFunction] {
              "calls": Array [
                Array [
                  true,
                ],
              ],
            },
            "placeholder": "Enter a name for this saved search",
            "quiet": true,
            "readOnly": false,
            "required": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Checkbox
            checked={false}
            indeterminate={false}
            label="Track this search result in dashboard"
            onChange={[Function]}
          />,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": false,
            "indeterminate": false,
            "label": "Track this search result in dashboard",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Label
            labelFor="Save This Search"
            value="Name"
          >
            <Textfield
              className={undefined}
              data-test="save-this-search-name-field"
              disabled={false}
              invalid={false}
              onChange={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      true,
                    ],
                  ],
                }
              }
              placeholder="Enter a name for this saved search"
              quiet={true}
              readOnly={false}
              required={false}
            />
          </Label>,
          <div
            className={undefined}
          >
            <Checkbox
              checked={false}
              indeterminate={false}
              label="Track this search result in dashboard"
              onChange={[Function]}
            />
          </div>,
          false,
        ],
        "data-test": "save-this-search-dialog-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Textfield
              className={undefined}
              data-test="save-this-search-name-field"
              disabled={false}
              invalid={false}
              onChange={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      true,
                    ],
                  ],
                }
              }
              placeholder="Enter a name for this saved search"
              quiet={true}
              readOnly={false}
              required={false}
            />,
            "labelFor": "Save This Search",
            "value": "Name",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": undefined,
              "data-test": "save-this-search-name-field",
              "disabled": false,
              "invalid": false,
              "onChange": [MockFunction] {
                "calls": Array [
                  Array [
                    true,
                  ],
                ],
              },
              "placeholder": "Enter a name for this saved search",
              "quiet": true,
              "readOnly": false,
              "required": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Checkbox
              checked={false}
              indeterminate={false}
              label="Track this search result in dashboard"
              onChange={[Function]}
            />,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "indeterminate": false,
              "label": "Track this search result in dashboard",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
